bundles/org.openhab.binding.echonetlite/src/main/java/org/openhab/binding/echonetlite/internal/EchonetLiteBridgeHandler.java

```diff
@@ -0,0 +1,401 @@
+            logger.debug("Received request: {}", message);
+            if (message instanceof NewDeviceMessage) {
+                newDeviceInternal((NewDeviceMessage) message);
+            } else if (message instanceof RefreshMessage) {
+                refreshDeviceInternal((RefreshMessage) message);
+            } else if (message instanceof RemoveDevice) {
+                removeDeviceInternal((RemoveDevice) message);
+            } else if (message instanceof UpdateDevice) {
+                updateDeviceInternal((UpdateDevice) message);
+            } else if (message instanceof StartDiscoveryMessage) {
+                startDiscoveryInternal((StartDiscoveryMessage) message);
+            } else if (message instanceof StopDiscoveryMessage) {
+                stopDiscoveryInternal((StopDiscoveryMessage) message);
+            }
+        }
+    }
+
+    private @Nullable Message requestsPoll() {
+        return requests.poll();
+    }
+
+    private void pollNetwork(EchonetChannel echonetChannel) {
+        try {
+            echonetChannel.pollMessages(echonetMessage, this::onMessage,
+                    EchonetLiteBindingConstants.NETWORK_WAIT_TIMEOUT);
+        } catch (IOException e) {
+            logger.warn("Failed to poll for messages", e);
+        }
+    }
+
+    private void onMessage(final EchonetMessage echonetMessage) {
+        final EchonetClass echonetClass = echonetMessage.sourceClass();
+        final SocketAddress sourceAddress = echonetMessage.sourceAddress();
+        if (null == echonetClass) {
+            logger.warn("Unable to find echonetClass for message: {}", echonetMessage.toDebug());
+            return;
+        }
+
+        if (null == sourceAddress) {
+            logger.warn("Unable to find sourceAddress for message: {}", echonetMessage.toDebug());
+            return;
+        }
+
+        final InstanceKey instanceKey = new InstanceKey((InetSocketAddress) sourceAddress, echonetClass,
+                echonetMessage.instance());
+        final Esv esv = echonetMessage.esv();
+
+        EchonetObject echonetObject = devicesByKey.get(instanceKey);
+        if (null == echonetObject) {
+            echonetObject = devicesByKey.get(discoveryKey);
+        }
+
+        logger.debug("Message {} for: {}", esv, echonetObject);
+        if (null != echonetObject) {
+            echonetObject.applyHeader(esv, echonetMessage.tid());
+            while (echonetMessage.moveNext()) {
+                final int epc = echonetMessage.currentEpc();
+                final int pdc = echonetMessage.currentPdc();
+                ByteBuffer edt = echonetMessage.currentEdt();
+                echonetObject.applyProperty(instanceKey, esv, epc, pdc, edt);
+            }
+        }
+    }
+
+    private void poll() {
+        try {
+            doPoll();
+            updateStatus(ThingStatus.ONLINE);
+
+            while (!Thread.currentThread().isInterrupted()) {
+                doPoll();
```

Comment:
Should status be updated to **ONLINE** here as well, or is it no possible to go **OFFLINE** within this loop?

Reply:
For the bridge, no.  Once the socket is bound then the bridge is online.